tests/subsys/fs/zms/src/main.c

```diff
@@ -0,0 +1,882 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This test is designed to be run using flash-simulator which provide
+ * functionality for flash property customization and emulating errors in
+ * flash operation in parallel to regular flash API.
+ * Test should be run on qemu_x86, native_sim target.
+ */
+
+#if !defined(CONFIG_BOARD_QEMU_X86) && !defined(CONFIG_ARCH_POSIX)
+#error "Run only on qemu_x86 or a posix architecture based target (for ex. native_sim)"
+#endif
+
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/ztest.h>
+
+#include <zephyr/drivers/flash.h>
+#include <zephyr/fs/zms.h>
+#include <zephyr/stats/stats.h>
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/sys/crc.h>
+#include "zms_priv.h"
+
+#define TEST_ZMS_FLASH_AREA        storage_partition
```

Comment:
I hope we can standardize on more open terms like `NVM` instead of reiterating `FLASH` for likely historical reasons. The addition of ZMS should not be tied to any specific type of non volatile memory technology...

Reply:
Indeed, ZMS is not tied to any specific type of volatile or non volatile memory but as its name indicates (Zephyr Mempry Storage) it is for any Storage device.
In this case, tests are made to work on flash_simulator which is a flash.
I can change these defines by removing FLASH refrence (TEST_ZMS_FLASH_AREA => TEST_ZMS_AREA).
But the references to flash will still be there across the file because we are using flash_simulator and flash_api